pulsarlib/messaging.go

```diff
@@ -392,6 +392,75 @@ func CreateConsumer(tenantID string, namespace string, topics []string, subscrip
 	return consumer, nil
 }
 
+/*
+	This API will create a Consumer for a particular topic.
+	The handler passed should implement the Handler interface from this module.
+	The consumer will create the subscription and be in a passive state until Start() is called.
+	The consumer can be Paused and Unpaused at any point.
+	The commitInterval used to commit messages after every n messages are consumed.
+	The Pause() function will flushout the already received messages and pause receiving any further messages.
+	The Unpause() function will resume receiving messages.
+	The Stop() function will flush existing messages and stop the consumer. It won't delete the subscription.
+	The Unsubscribe() function can be used if subscription needs to be deleted.
+	The Stats() function provides the stats for messages consumed.
+
+	Creating multiple instances of Consumer for same topic will deliver message to only one of the instances.
+	Inorder to recreate a Consumer for same topic make sure Stop() is called on old Consumer instance.
+    retryEnabled will let the consumer retry message in case of HandleMessage return `RetryMessage` struct.
+*/
+func CreateSingleTopicConsumer(tenantID string, namespace string, topic string, subscriptionName string, handler Handler, commitInterval uint64, retryEnabled bool) (Consumer, error) {
```

Comment:
Let us deprecate commitInterval

Reply:
then when should we commit message?

Reply:
Our commit on consumer is currently a NoOp. We can remove it altogether.

Reply:
got it